python/ray/autoscaler/autoscaler.py

```diff
@@ -426,16 +434,14 @@ def can_update(self, node_id):
             return False
         return True
 
-    def launch_new_node(self, count, instance_type=None):
+    def launch_new_node(self, count, instance_config=None):
```

Comment:
Why are we passing the whole config instead of instance type here? Can't we keep the instance type, and just pull the instance config out at the last minute, so it doesn't need to be plumbed all the way through everything?

Reply:
We need to full config in order to track the resources that the instance is started with (i.e. `p3.2_no_cpu`). We will also need to the config to help track max_workers, etc

Reply:
Can't you just look up the full config later on? Only need to pass the key around.